docs/01-app/02-guides/json-ld.mdx

```diff
@@ -6,7 +6,11 @@ description: Learn how to add JSON-LD to your Next.js application to describe yo
 
 [JSON-LD](https://json-ld.org/) is a format for structured data that can be used by search engines and AI to to help them understand the structure of the page beyond pure content. For example, you can use it to describe a person, an event, an organization, a movie, a book, a recipe, and many other types of entities.
 
-Our current recommendation for JSON-LD is to render structured data as a `<script>` tag in your `layout.js` or `page.js` components. For example:
+Our current recommendation for JSON-LD is to render structured data as a `<script>` tag in your `layout.js` or `page.js` components.
+
+The following snippet uses `JSON.stringify`, which does not sanitize malicious strings used in XSS injection. To prevent this type of vulnerability, you can scrub `HTML` tags from the `JSON-LD` payload, for example, by replacing the character, `<`, with its unicode equivalent, `\u003c`.
+
+Review your organization's recommended approach to sanitize potentially dangerous strings, or use community maintained alternatives for `JSON.stringify` such as, [serialize-javascript](https://www.npmjs.com/package/serialize-javascript).
```

Comment:
this could also recommend https://github.com/google/react-schemaorg, which does all the necessary escaping according to spec.

Reply:
Nice, I didn't know about that package, and it seems to play well with `schema-dts` too.

Reply:
> it seems to play well with schema-dts too

i think both packages are maintained by the same team.